ocaml/xapi/xapi.ml

```diff
@@ -531,6 +531,7 @@ let start_redo_log () =
       && not (bool_of_string (Localdb.get Constants.ha_armed))
     then (
       debug "Redo log was enabled when shutting down, so restarting it" ;
+      Static_vdis.reattempt_on_boot_attach () ;
```

Comment:
Is there a way to share init code in both cases instead of adding more actions here and comparing with the HA case?

Reply:
This is pretty minimal; otherwise we would need to refactor all this redo log code. I suspect this fix is missing in the DR code as well.